logstash-core/src/main/java/org/logstash/instrument/metrics/MetricKeys.java

```diff
@@ -39,10 +39,61 @@ private MetricKeys() {
 
     public static final RubySymbol IN_KEY = RubyUtil.RUBY.newSymbol("in");
 
-    public static final RubySymbol DURATION_IN_MILLIS_KEY =
-        RubyUtil.RUBY.newSymbol("duration_in_millis");
+    public static final RubySymbol PLUGINS_KEY = RubyUtil.RUBY.newSymbol("plugins");
+
+    public static final RubySymbol INPUTS_KEY = RubyUtil.RUBY.newSymbol("inputs");
+
+    // DLQ related metric keys
+    public static final RubySymbol DURATION_IN_MILLIS_KEY = RubyUtil.RUBY.newSymbol("duration_in_millis");
+
+    public static final RubySymbol PUSH_DURATION_KEY = RubyUtil.RUBY.newSymbol("queue_push_duration_in_millis");
+
+    public static final RubySymbol PAGE_CAPACITY_IN_BYTES_KEY = RubyUtil.RUBY.newSymbol("page_capacity_in_bytes");
+
+    public static final RubySymbol MAX_QUEUE_SIZE_IN_BYTES_KEY = RubyUtil.RUBY.newSymbol("max_queue_size_in_bytes");
+
+    public static final RubySymbol MAX_QUEUE_UNREAD_EVENTS_KEY = RubyUtil.RUBY.newSymbol("max_unread_events");
+
+    public static final RubySymbol QUEUE_SIZE_IN_BYTES_KEY = RubyUtil.RUBY.newSymbol("queue_size_in_bytes");
+
+    public static final RubySymbol FREE_SPACE_IN_BYTES_KEY = RubyUtil.RUBY.newSymbol("free_space_in_bytes");
+
+    public static final RubySymbol STORAGE_TYPE_KEY = RubyUtil.RUBY.newSymbol("storage_type");
+
+    public static final RubySymbol PATH_KEY = RubyUtil.RUBY.newSymbol("path");
+
+    public static final RubySymbol TYPE_KEY = RubyUtil.RUBY.newSymbol("type");
+
+    public static final RubySymbol QUEUE_KEY = RubyUtil.RUBY.newSymbol("queue");
+
+    public static final RubySymbol DLQ_KEY = RubyUtil.RUBY.newSymbol("dlq");
+
+    public static final RubySymbol STORAGE_POLICY_KEY = RubyUtil.RUBY.newSymbol("storage_policy");
+
+    public static final RubySymbol DROPPED_EVENTS_KEY = RubyUtil.RUBY.newSymbol("dropped_events");
+
+    public static final RubySymbol EXPIRED_EVENTS_KEY = RubyUtil.RUBY.newSymbol("expired_events");
+
+    public static final RubySymbol LAST_ERROR_KEY = RubyUtil.RUBY.newSymbol("last_error");
 
     public static final RubySymbol FILTERED_KEY = RubyUtil.RUBY.newSymbol("filtered");
 
     public static final RubySymbol STATS_KEY = RubyUtil.RUBY.newSymbol("stats");
+
+    // Flow metric keys
+    public static final RubySymbol FLOW_KEY = RubyUtil.RUBY.newSymbol("flow");
+
+    public static final String INPUT_THROUGHPUT_KEY = "input_throughput";
+
+    public static final String OUTPUT_THROUGHPUT_KEY = "output_throughput";
+
+    public static final String FILTER_THROUGHPUT_KEY = "filter_throughput";
+
+    public static final String QUEUE_BACKPRESSURE_KEY = "queue_backpressure";
+
+    public static final String WORKER_CONCURRENCY_KEY = "worker_concurrency";
+
+    public static final String UPTIME_IN_SECONDS_KEY = "uptime_in_seconds";
+
+    public static final String UPTIME_IN_MILLIS_KEY = "uptime_in_millis";
```

Comment:
NIT: The rest of the entries here are `RubySymbol`, and static-final-ing them as `String` here means at runtime we need to repeatedly symbolize them. Since `RubySymbol#asJavaString` allows us to efficiently get a Java `String`, I would prefer these be `RubySymbol` to match the others in the `MetricKeys` namespace.

Reply:
Separately, `MetricKeys` and all of the entries ending in `_KEY` is a bit redundant. Perhaps this is to make it easier to see what they are and less likely to collide when we wildcard static import `MetricKeys.*`, in which case I would recommend wildcard importing so we don't have to repeatedly have the overly-descriptive `MetricKeys.SOME_DESCRIPTIVE_KEY` throughout the codebase.

Reply:
That made me thought for a while since I didn't need `RubySymbol`: should we avoid using JRuby definitions for long term to move to plain Java?! But as you mentioned getting java string with `asJavaString()` doesn't compile anything at runtime, so we are safe to use any option.